src/DCG/main.py

```diff
@@ -110,28 +110,41 @@ def cast_label_to_one_hot_and_prototype(self, y_labels_batch, return_prototype=T
             return y_one_hot_batch
 
 
-def train_DCG(dcg, params, train_loader):
+def train_DCG(dcg, params, train_loader, test_loader):
 
     criterion = nn.CrossEntropyLoss()
     brier_score = nn.MSELoss()
     optimizer = torch.optim.SGD(dcg.parameters(), lr=0.1, weight_decay=1e-4, momentum=0.9)
     dcg.train()
-    loss_epoch = []
     pretrain_start_time = time.time()
+    test_iter = iter(test_loader)
+    loss_batch, loss_test_array = [], []
     for epoch in range(params["num_epochs"]):
-        loss_batch = []
+        # loss_batch = []
         for feature_label_set in train_loader:
+            # train loss
+            dcg.train()
             x_batch, y_labels_batch = feature_label_set
             y_one_hot_batch, y_logits_batch = dcg.cast_label_to_one_hot_and_prototype(y_labels_batch)
             aux_loss = dcg.nonlinear_guidance_model_train_step(criterion, x_batch, y_one_hot_batch, optimizer)
             loss_batch.append(aux_loss)
+            # eval loss
```

Comment:
My assumption was that you will create a separate eval function for DCG

Reply:
I don’t think there is any need to create separate eval function. I am just getting eval loss at each train iteration.

Reply:
Ohh it's supposed to be validation. Do you think it will be better to create a val_loader instead of using test_loader?